app/scripts/ui.js

```diff
@@ -40,10 +41,29 @@ const PHISHING_WARNING_PAGE_TIMEOUT = ONE_SECOND_IN_MILLISECONDS;
 const PHISHING_WARNING_SW_STORAGE_KEY = 'phishing-warning-sw-registered';
 
 let extensionPort;
+let traceContext;
```

Comment:
This object is probably lightweight, but even so, this looks like it might "leak". Once we are done using it I worry that it is still "reachable" and not eligible for being garbage collected. Should we pass this to methods that need it so it so we can be sure it can be garbage collected?

Note: I could be wrong about how variables in the module scope work as far as GC in concerned. This might be perfectly fine.